src/seedu/addressbook/Main.java

```diff
@@ -9,121 +9,161 @@
+			 * unchecked exception here is because the caller cannot reasonably
+			 * be expected to recover from an exception. Cf
+			 * https://docs.oracle.com/javase/tutorial/essential/exceptions/
+			 * runtime.html
+			 * =================================================================
+			 * ======================================
+			 */
+			throw new RuntimeException(e);
+		}
+	}
+
+	/** Prints the Goodbye message and exits. */
+	private void exit() {
+		ui.showGoodbyeMessage();
+		System.exit(0);
+	}
+
+	/**
+	 * Reads the user command and executes it, until the user issues the exit
+	 * command.
+	 */
+	private void runCommandLoopUntilExitCommand() {
+		Command command;
+		try {
+			do {
+				String userCommandText = ui.getUserCommand();
+				checkFileExist();
+				command = new Parser().parseCommand(userCommandText);
+				CommandResult result = executeCommand(command);
+				recordResult(result);
+				ui.showResultToUser(result);
+
+			} while (!ExitCommand.isExit(command));
+			
+		} catch (FileNotFoundException ERROR_MESSAGE) {
+			System.out.println(ERROR_MESSAGE);
+			System.exit(0);
+		}
+	}
+
+	/**
+	 * This method checks if the addressbook.txt file has been deleted 
+	 * by the user while the program is still running.
+	 * @throws FileNotFoundException
+	 */
+	private void checkFileExist() throws FileNotFoundException {
+		final String ERROR_MESSAGE = "File has been deleted, exiting program.";
+		File addressBook = new File("addressbook.txt");
+		if (!addressBook.exists()){
+			throw new FileNotFoundException(ERROR_MESSAGE);
+		}
+	}
+
+	/**
+	 * Updates the {@link #lastShownList} if the result contains a list of
+	 * Persons.
+	 */
+	private void recordResult(CommandResult result) {
+		final Optional<List<? extends ReadOnlyPerson>> personList = result.getRelevantPersons();
+		if (personList.isPresent()) {
+			lastShownList = personList.get();
+		}
+	}
+
+	/**
+	 * Executes the command and returns the result.
+	 * 
+	 * @param command
+	 *            user command
+	 * @return result of the command
+	 */
+	private CommandResult executeCommand(Command command) {
+		try {
+			command.setData(addressBook, lastShownList);
+			CommandResult result = command.execute();
+			storage.save(addressBook);
+			return result;
+		} catch (Exception e) {
+			ui.showToUser(e.getMessage());
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Creates the StorageFile object based on the user specified path (if any)
+	 * or the default storage path.
+	 * 
+	 * @param launchArgs
+	 *            arguments supplied by the user at program launch
+	 * @throws InvalidStorageFilePathException
+	 *             if the target file path is incorrect.
+	 */
+	private StorageFile initializeStorage(String[] launchArgs) throws InvalidStorageFilePathException {
+		boolean isStorageFileSpecifiedByUser = launchArgs.length > 0;
+		return isStorageFileSpecifiedByUser ? new StorageFile(launchArgs[0]) : new StorageFile();
+	}
```

Comment:
Avoid auto-formatting unrelated parts of the code. Doing so can result in unrelated changes being included in the commit. You can use sourcetree to look through changes before committing, and discard any unrelated changes.